### DISCUSSION TITLE
Channel update taking multiple frameworks with the same identifier & code

### DISCUSSION THREAD

**Discussion by vinukumar-vs** [Sunbird-Knowlg/Community#109] [Bugs]
While setting up the framework & channel, I have noticed an issue with channel having the duplicate frameworks with the same identifier & code.

`curl --location -g --request PATCH '{{host}}/api/channel/v1/update/{{channel_id}}' \
--header 'Content-Type: application/json' \
--header 'Authorization: {{kong_api_key}}' \
--header 'x-channel-id: {{channel_name}}' \
--data '{
    "request": {
        "channel": {
            "frameworks": [
                {
                    "identifier": "NCF",
                    "name": "NCF"
                }
            ]
        }
    }
}'`

In channel read I can see multiple frameworks with the same code & identifiers(but descriptions are different)

<img width="1081" alt="Screenshot 2023-03-20 at 4 08 15 PM" src="https://user-images.githubusercontent.com/4496096/226315879-b916542a-8b1a-4b4c-9730-857292234971.png">

I am using channel update API to remove the duplicates, But I am not able to do. Any suggestion how can we delete it other than removing directly from DB.

Few questions:

- How it is taking different frameworks with the same identifier?
- Framework create allows duplicates with the same Identifier?

@Krishnaj20 @pallakartheekreddy

**Comment by pallakartheekreddy** [comment]
@vinukumar-vs please refer below thread. it is related to same issue.
https://github.com/Sunbird-Knowlg/Community/discussions/102#discussioncomment-5249944
